### DISCUSSION TITLE
Reload badges

### DISCUSSION THREAD

**Discussion by ScrubN** [Chatterino/chatterino2#5480] [Ideas] [answered]
Within the last two or so weeks, 2 streamers that I watch updated their badges while live on stream. Unfortunately it seems Chatterino does not feature a way to manually update channel badges without closing the tab or reloading the application. 

It would be nice if there was a button to reload badges on the fly. Perhaps it could be placed under "Reload subscriber emotes" in the context menu?

**Comment by Felanbird** [comment, marked as answer]
I feel like this has been discussed before and badges objectively do not change often enough to need a reload button.

Closing the tab and re-opening it with `ctrl+W` & `ctrl+G` takes actually half of a second*

**Reply by Felanbird** [reply]
Also when Chatterino homies added a badge reloading feature they caused 7TV a huge incident due to the way it was done lol, so there's that for 3rd party badge reloading

**Reply by ScrubN** [reply]
> I feel like this has been discussed before and badges objectively do not change often enough to need a reload button.
> 
> Closing the tab and re-opening it with `ctrl+W` & `ctrl+G` takes actually half of a second*

It seemed that the argument was more about if it should have a dedicated button or not (#3118), however I do agree that closing and reopening the tab could be sufficient.

**Reply by ScrubN** [reply]
> Also when Chatterino homies added a badge reloading feature they caused 7TV a huge incident due to the way it was done lol, so there's that for 3rd party badge reloading

Yeah I guess reloading badges could be very server heavy.

I just remembered a moment when Theo t3 mentioned how adding badges to the mod view caused an internal DDOS against the Twitch payment service due to how paid badges are(/were?) authenticated. Maybe adding a refresh button isn't a great idea after all 😅

**Reply by MattIPv4** [reply]
👋 I would like to push back on this and the only reason for needing to reload is that badges changing -- I often run into Chatterino simply not having badges for a channel (or having partial badges; not sure why) and would greatly appreciate being able to reload them in the same way I can reload emotes or reconnect to the chat.
